src/main/java/net/neoforged/neoforge/server/command/TPSCommand.java

```diff
@@ -60,4 +60,65 @@ private static long mean(long[] values) {
             sum += v;
         return sum / values.length;
     }
+
+    private static Component createComponent(MinecraftServer server, @Nullable ServerLevel dimension) {
+        MutableComponent prefix;
+        long[] times;
+        TickRateManager tickRateManager;
+
+        if (dimension == null) {
+            prefix = Component.translatable("commands.neoforge.tps.summary.overall");
+            times = server.getTickTimesNanos();
+            tickRateManager = server.tickRateManager();
+        } else {
+            var dimKey = dimension.dimension();
+            var dimensionTimes = server.getTickTime(dimKey);
+
+            prefix = Component.translatable("commands.neoforge.tps.summary.dimension", createDimensionComponent(dimension));
+            times = dimensionTimes == null ? UNLOADED : dimensionTimes;
+            tickRateManager = dimension.tickRateManager();
+        }
+
+        var tickTime = mean(times) * 1.0E-6D;
```

Comment:
This line can be improved by first using the `Stats.meanOf` method from Guava, replacing the self-made `mean` method, and replacing the `* 1.0E-6D` part with `/ TimeUtil.NANOSECONDS_PER_MILLISECOND`.

The second change removes the magic constant to use a named constant instead, which makes the code more understandable, and in my previous testing avoided some possible floating-point rounding errors.